include/ANurbs/src/Grid.h

```diff
@@ -74,6 +75,8 @@ class Grid
         const int& index
     )
     {
+        assert(index < m_values.size() && "Index out of range");
```

Comment:
KRATOS_ERROR_IF_NOT() ... ;)

Reply:
The idea was to use `assert` for better performance. I stopped this branch because I think it would be better to throw exceptions also in the release... anyway no `KRATOS_ERROR_IF_NOT` here ;)